WordPress/WordPress.xcodeproj/project.pbxproj

```diff
@@ -16581,6 +16591,177 @@
 			shellScript = "$SRCROOT/../Scripts/BuildPhases/GenerateCredentials.sh\n";
 			showEnvVarsInLog = 0;
 		};
+		09DBEA4C281329ED0019724E /* [Lint] Check AppLocalizedString usage */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+			);
+			name = "[Lint] Check AppLocalizedString usage";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = "/bin/zsh -euo pipefail";
```

Comment:
Using `zsh` as the shell here (instead of `sh`) intentionally, because:
 - This calls a Ruby script
 - Which thus depends on the user's configuration of `rbenv` or `rvm` or similar
 - And those tools are usually set up in `~/.zshrc`

So using `/bin/sh` here would not read the user's `~/.zshrc`, thus not have `rbenv` properly set up, and thus might not use the expected version of Ruby or fail.

Reply:
> Or... could we use `/usr/bin/env ruby` directly? 🤔

No, that doesn't make sense because then the content of the `shellScript` would have to be actual Ruby code, but we need it DRY and in a script. (I tried that [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/18444/commits/a4aef35a68817d50ff3509b323000636719185de)...)

**I think a next step might be to wrap the Ruby script call into a `.sh` script that uses `bundle exec ruby ...` and does all the necessary checks beforehand.**

Reply:
Ahhh dammit I was afraid such case and portability issue would happen 😭 
Not the first time I encounter the question of "How to make a script called by Xcode use the user's shell setup, especially so it knows about Ruby and Bundler". This is tricky because:

 - We should probably not assume which shell the users use (like I did here with `zsh`, considering that because `zsh` was now the default on macOS, this would be a safe bet… but that discards anyone who decided to use an alternate shell like `fish` indeed.
    - Though to be honest personally if I were to use `fish` I'd probably organize my rc files so that `zsh` still works and configure `ruby` and `bundler`, and then my fish rc would `source` that `~/.zshrc` (and possibly add fish-specific stuff after that), that way `zsh` would still work as well even if my everyday shell were `fish`…
    - …but eh, that's another assumption I can't really make to ensure this is really portable
 - We can't use `/usr/bin/env ruby` either here, because that risks not only using the system ruby, but also not using `bundler` (and thus picking up the `xcodeproj` gem installed in the project's bundle, as part of a cocoapods dependency anyway)
    - To be fair my current solution didn't call `bundler` either, as you noticed. That worked on my setup because I have a plugin configured in `~/.zshrc` that automatically prepends `bundle exec` to many known gems and executions or ruby scripts by default for me, which is probably why I missed that when working on my PR
    - But that's obviously yet again not an assumption we can make
 - I'll try your suggestion of making the Script Build Phase use `/bin/sh -euo pipefail`, and it just calling a `.sh` script that would wrap the calls to `bundle exec` and to the ruby script.
    - But tbh I'm not super confident that this will work perfectly either, because the `.sh` shell spawned by Xcode is a child process of Xcode, not of the login shell and Terminal, so it would not depend on (nor inherit the setup of) the user's default shell (the one they use when launching their `Terminal.app`). Such a child `sh` shell spawned by Xcode would thus not have knowledge of the setup and config related to `bundler` and `rbenv` the user usually have for their default shell in Terminal…
    - And obviously we can't either make the our `.sh` wrapper script assume anything about the user's setup; so the potential ideas of configuring `rbenv` from that wrapper script (e.g. calling `eval $(rbenv init -)` or `export PATH=~/.rbenv/shims:$PATH` etc) won't work either, as users might be using `rvm` instead of `rbenv` in their system…

I've found [this post](https://mgrebenets.github.io/xcode/2019/04/04/xcode-build-phases-and-environment) which explains the issue (which confirms what I've encountered with this kind of cases in the past), and also provides some ideas… but it also relies on the assumption that the user a particular Ruby Version manager (`rvm` in that post's case)… so still not a universal solution.

Maybe I'll have to make my wrapper script just test which Ruby Version Manager tool is installed (test `which rbenv` / `command -v rbenv` vs `command -v rvm` and act accordingly) and cover both cases… (and here goes my plan for the day 😅 )

Reply:
@mokagio It took me most of the day to figure out a working (?) solution that should cover cases of folks using any shell, and folks using either `rbenv` or `rvm`, but the last 2 commits should finally work.

At least @jhnstn helped testing the setup of using `rvm`, and @twstokes helped testing the setup of using `rbenv` (see Slack thread in p1651001863417419/1650990034.879279-slack-C011BKNU1V5) and after a lot of trial and error tweaking my script to make it work, this finally seem to cover all setups properly 🤞